source/FileFormats/GBL.vb

```diff
@@ -0,0 +1,227 @@
+        ''' Gibt an, ob beim Import des Dateiformats der Importdialog angezeigt werden soll
+        ''' </summary>
+        Public Overrides ReadOnly Property UseImportDialog() As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Possible data types of values
+        ''' </summary>
+        Private Enum DataTypes As Integer
+            [Integer] = 1
+            [Single] = 2
+            [Double] = 3
+            [Boolean] = 4
+        End Enum
+
+        ''' <summary>
+        ''' Lengths in bytes for different data types
+        ''' </summary>
+        Private Shared ReadOnly DataTypeLengths As New Dictionary(Of DataTypes, Integer) From {
+        {DataTypes.Integer, 4},
+        {DataTypes.Single, 4},
+        {DataTypes.Double, 8},
+        {DataTypes.Boolean, 1}
+    }
+
+        ''' <summary>
+        ''' The data type of values in this file instance
+        ''' </summary>
+        Private DataType As DataTypes
+
+        ''' <summary>
+        ''' The length in bytes of the values in this file instance
+        ''' </summary>
+        ''' <returns></returns>
+        Private ReadOnly Property DataTypeLength As Integer
+            Get
+                Return GBL.DataTypeLengths(Me.DataType)
+            End Get
+        End Property
+
+        Private Shared ReadOnly ColumnNames As String() = {
+            "Qzu", "Qab", "Pges_zu", "Pges_ab",
+            "AFS_zu", "AFS_ab", "AFS63_zu", "AFS63_ab", "NH4-N_zu", "NH4-N_ab",
+            "CSB_zu", "CSB_ab", "T_zu", "T_ab", "pH_zu",
+            "pH_ab", "O2_zu", "O2_ab", "NH3_zu", "k2", "tf"
+        }
+
+        Private Shared ReadOnly ColumnUnits As String() = {
+            "l/s", "l/s", "mg/l", "mg/l",
+            "mg/l", "mg/l", "mg/l", "mg/l", "mg/l", "mg/l",
+            "mg/l", "mg/l", "°C", "°C", "-",
+            "-", "mg/l", "mg/l", "mg/l", "-", "min"
+        }
+
+        Public Sub New(FileName As String, Optional ReadAllNow As Boolean = False)
+
+            MyBase.New(FileName)
+
+            'Voreinstellungen
+            Me.Dateformat = Helpers.CurrentDateFormat 'irrelevant because binary
+            Me.iLineData = 0
+            Me.UseUnits = True
+
+            Call Me.readSeriesInfo()
+
+            If (ReadAllNow) Then
+                'Direkt einlesen
+                Call Me.selectAllSeries()
+                Call Me.readFile()
+            End If
+
+        End Sub
+
+        ''' <summary>
+        ''' Reads series info for fixed GBL format structure
+        ''' </summary>
+        Public Overrides Sub readSeriesInfo()
+
+            Me.TimeSeriesInfos.Clear()
+
+            ' GBL format has fixed structure: 8 bytes date + 21 * 4 bytes data columns
+            Me.DataType = DataTypes.Single ' Fixed data type for GBL format
+
+            ' Create 21 time series info objects for the fixed columns
+            For i As Integer = 0 To 20 ' 21 columns (0-20)
```

Comment:
suggestion: loop over the length of `ColumnNames` instead of a constant number. If `ColumnNames` ever changes, this loop will automatically adapt accordingly.